### PR TITLE
Gh-3234: TypeSubTypeValue Seeded Query Support in GafferPop

### LINKED ISSUE #3234: Support TSTV seeded queries in GafferPop
**Describe the new feature you'd like**
`TypeSubTypeValue` is a data type that can be used for IDs in Gaffer. We need to ensure that users can use TSTVs via Gremlin using GafferPop.

**Additional context**
The simplest way we could allow it would be treating them as String IDs in the form `myType|mySubType|myValue` and then parse them as `TypeSubTypeValues` once they've got through the server layer. Essentially so a user could do a seeded Gremlin query like: `g.V("myType|mySubType|myValue")`.

### PR DESCRIPTION
**Pull request by tb06904**
Adds basic support for seeding queries in GafferPop with TypeSubTypeValues by parsing specific string formats separated via a `|` to construct the correct object to seed the query with. 

As an example `g.V("t:type|st:sub|v:value")` -> `new TypeSubTypeValue("type", "sub", "value")`

# Related issue

- Resolve #3234

### CODE REVIEW

File: library/tinkerpop/src/main/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopGraph.java [hunk 1]
```diff
@@ -980,18 +981,40 @@ private List<ElementSeed> getElementSeeds(final Iterable<Object> ids) {
...
+     * @param id The ID.
+     * @return The ID as its relevant type.
+     */
+    private Object getIDAsRelevantType(final Object id) {
+        if (id instanceof String) {
+            // See if can split into a TSTV
+            String[] split = ((String) id).split("\\|");
+            if (split.length == 3) {
+                LOGGER.debug("Parsing ID as a TSTV: {}", id);
+                return new TypeSubTypeValue(split[0], split[1], split[2]);
```
Comment (p29876) [inline library/tinkerpop/src/main/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopGraph.java:1012]: May be worth changing to regex to bring it in line style-wise with #3230 
But this is still very readable so happy either way

**Review by p29876 — APPROVED** [review]

File: library/tinkerpop/src/main/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopGraph.java (hunk 1 above)
Comment (p29876) [reply library/tinkerpop/src/main/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopGraph.java:1012]: assuming we don't have to handle any whitespace here?

**Review by cn337131 — DISMISSED** [review]

File: library/tinkerpop/src/main/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopGraph.java (hunk 1 above)
Comment (tb06904) [reply library/tinkerpop/src/main/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopGraph.java]: Updated to use a regex instead with a slightly tweaked format. I don't think we really want to handle white space as we can't know if that is meant to be part of the type, subtype or value so safer to take whats between the `|` delimiter as literal.

**Review by p29876 — DISMISSED** [review]

File: library/tinkerpop/src/main/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopGraph.java [hunk 2]
```diff
@@ -245,6 +248,7 @@ public enum DefaultIdManager {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(GafferPopGraph.class);
     private static final String GET_ALL_DEBUG_MSG = "Requested a GetAllElements, results will be truncated to: {}.";
+    private static final Pattern TSTV_REGEX =  Pattern.compile("^tstv:(?<type>[A-Za-z0-9-]*)\\|(?<stype>[A-Za-z0-9-]*)\\|(?<val>[A-Za-z0-9-]*)$");
```
Comment (cn337131) [inline library/tinkerpop/src/main/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopGraph.java:251]: just to check that TSTV's can't contain a pipe in the type/subtype/value ?

Comment (tb06904) [reply library/tinkerpop/src/main/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopGraph.java]: Updated the regex match to allow seeding like `g.V("t:thetype|st:thesubstype|v:theval")` this should hopefully avoid majority of conflicts

**Review by cn337131 — APPROVED** [review]

**Review by p29876 — APPROVED** [review]

**Pull request merged**